pkg/tsdb/cloudwatch/request_parser.go

```diff
@@ -195,52 +207,70 @@ func parseRequestQuery(model *simplejson.Json, refId string, startTime time.Time
 		}
 		id = fmt.Sprintf("query%s", suffix)
 	}
-	expression := model.Get("expression").MustString("")
-	sqlExpression := model.Get("sqlExpression").MustString("")
-	alias := model.Get("alias").MustString()
-	label := model.Get("label").MustString()
-	returnData := !model.Get("hide").MustBool(false)
-	queryType := model.Get("type").MustString()
-	timezoneUTCOffset := model.Get("timezoneUTCOffset").MustString("")
-
-	if queryType == "" {
+
+	returnData := false
+	if model.Hide != nil {
+		returnData = !*model.Hide
+	}
+
+	if model.QueryType == "" {
 		// If no type is provided we assume we are called by alerting service, which requires to return data!
 		// Note, this is sort of a hack, but the official Grafana interfaces do not carry the information
 		// who (which service) called the TsdbQueryEndpoint.Query(...) function.
 		returnData = true
 	}
 
-	matchExact := model.Get("matchExact").MustBool(true)
-	metricQueryType := metricQueryType(model.Get("metricQueryType").MustInt(0))
-
 	var metricEditorModeValue metricEditorMode
-	memv, err := model.Get("metricEditorMode").Int()
-	if err != nil && len(expression) > 0 {
+	if model.MetricEditorMode == nil && len(model.Expression) > 0 {
 		// this should only ever happen if this is an alerting query that has not yet been migrated in the frontend
 		metricEditorModeValue = MetricEditorModeRaw
 	} else {
-		metricEditorModeValue = metricEditorMode(memv)
+		if model.MetricEditorMode != nil {
+			metricEditorModeValue = metricEditorMode(*model.MetricEditorMode)
+		} else {
+			metricEditorModeValue = metricEditorMode(0)
+		}
+	}
+
+	statValue := ""
```

Comment:
You could create the return object here and set values on the return object instead of creating these intermediate values.